artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ChannelImpl.java

```diff
@@ -380,7 +380,7 @@ public Packet sendBlocking(final Packet packet,
 
             connection.getTransportConnection().write(buffer, false, false);
 
-            long toWait = connection.getBlockingCallTimeout();
```

Comment:
Looking at the code, I can't see how connection will ever be null here..  I think you should remove this one...

For instance... the very same connection was used before within a lock.. what would issue a NPE a lot earlier than here.

I don't see how connection could ever be null.